templates/history-purchases.php

```diff
@@ -8,38 +8,46 @@
  * This template is used to display the purchase history of the current user.
  */
 if ( is_user_logged_in() ):
-	$purchases = edd_get_users_purchases( get_current_user_id(), 20, true, 'any' );
-	if ( $purchases ) :
+	$payments = edd_get_users_purchases( get_current_user_id(), 20, true, 'any' );
+	if ( $payments ) :
 		do_action( 'edd_before_purchase_history' ); ?>
 		<table id="edd_user_history" class="edd-table">
 			<thead>
 				<tr class="edd_purchase_row">
 					<?php do_action('edd_purchase_history_header_before'); ?>
 					<th class="edd_purchase_id"><?php _e('ID','easy-digital-downloads' ); ?></th>
 					<th class="edd_purchase_date"><?php _e('Date','easy-digital-downloads' ); ?></th>
-					<th class="edd_purchase_amount"><?php _e('Amount','easy-digital-downloads' ); ?></th>
+					<th class="edd_purchase_amount"><?php _e('Amount','easy-digital-dxownloads' ); ?></th>
 					<th class="edd_purchase_details"><?php _e('Details','easy-digital-downloads' ); ?></th>
 					<?php do_action('edd_purchase_history_header_after'); ?>
 				</tr>
 			</thead>
-			<?php foreach ( $purchases as $post ) : setup_postdata( $post ); ?>
-				<?php $purchase_data = edd_get_payment_meta( $post->ID ); ?>
+			<?php foreach ( $payments as $payment ) : ?>
+				<?php $payment = new EDD_Payment( $payment->ID ); ?>
 				<tr class="edd_purchase_row">
-					<?php do_action( 'edd_purchase_history_row_start', $post->ID, $purchase_data ); ?>
-					<td class="edd_purchase_id">#<?php echo edd_get_payment_number( $post->ID ); ?></td>
-					<td class="edd_purchase_date"><?php echo date_i18n( get_option('date_format'), strtotime( get_post_field( 'post_date', $post->ID ) ) ); ?></td>
+					<?php do_action( 'edd_purchase_history_row_start', $payment->ID, $payment->payment_meta ); ?>
+					<td class="edd_purchase_id">#<?php echo $payment->number ?></td>
+					<td class="edd_purchase_date"><?php echo date_i18n( get_option('date_format'), strtotime( $payment->date ) ); ?></td>
 					<td class="edd_purchase_amount">
-						<span class="edd_purchase_amount"><?php echo edd_currency_filter( edd_format_amount( edd_get_payment_amount( $post->ID ) ) ); ?></span>
+						<span class="edd_purchase_amount"><?php echo edd_currency_filter( edd_format_amount( $payment->total ) ); ?></span>
 					</td>
 					<td class="edd_purchase_details">
-						<?php if( $post->post_status != 'publish' ) : ?>
-						<span class="edd_purchase_status <?php echo $post->post_status; ?>"><?php echo edd_get_payment_status( $post, true ); ?></span>
-						<a href="<?php echo esc_url( add_query_arg( 'payment_key', edd_get_payment_key( $post->ID ), edd_get_success_page_uri() ) ); ?>">&raquo;</a>
+						<?php if( $payment->status != 'publish' ) : ?>
+							<span class="edd_purchase_status <?php echo $payment->status; ?>"><?php echo $payment->status_nicename; ?></span>
+							<?php if ( $payment->is_recoverable() ) : ?>
+								<?php
+								$query_args = array(
```

Comment:
Let's turn this into a function `edd_get_payment_recovery_url()` (or a method of EDD_Payment). 

That will make it much easier for devs to extend and for us to reuse it (such as on View Order Details).